autolab/docker
# SSL setup support

This adds SSL support by using Let's Encrypt to obtain a certificate. The main complexity is that before we obtain a real certificate, we need a dummy certificate, if not Nginx in the `autolab` container will not start. This is what the `init-letsencrypt.sh` does. I took a lot of inspiration from https://medium.com/@pentacent/nginx-and-lets-encrypt-with-docker-in-less-than-5-minutes-b4b8a60d3a71 for this setup, it is worth to have a quick glance through. 

This also makes an SSL setup the default option now.

To test:
See the updated README for instructions on how to run it

Future work:
Add auto-renewal of certificates by Certbot

![Screen Capture_select-area_20201028180207](https://user-images.githubusercontent.com/9707110/97501485-4bb2a400-19ac-11eb-94a0-09dfc351c015.png)
